### PR TITLE
Countdown in hours and minutes

### PR DESCRIPTION
**Pull request by salalo**
#22 calculate time left before market closes in different units

### CODE REVIEW

**Comment by salalo** [comment]
If you need `daysLeft: number` in the interface, just close. I saw that `daysLeft` was only used in one place to display the data.

**Comment by kacpergumieniuk** [comment]
Thanks @salalo  for the pull request, I'll do some review and point out potential problems, so you can improve the solution

File: utils/calculateMarketClosing.ts [hunk 1]
```diff
@@ -1,38 +1,70 @@
...
+          minutesDifference > 1 ? "s" : ""
+        } left`;
+      } else {
+        return "Market closed";
+      }
+    }
+  }
+
+  return message;
+};
```
Comment (kacpergumieniuk) [inline utils/calculateMarketClosing.ts:70]: I think we got poor readability here. Below you got my proposition, where we just check if some condition is met, and if yes we just return actual message and end the function. You can give feedback about my idea or just implement it.
```
const calculateTimeLeft = (targetDate: Date): string => {
  const currentDate = new Date();
  const daysDifference = differenceInDays(targetDate, currentDate);

  if (daysDifference > 0) {
    return `${daysDifference} day${daysDifference > 1 ? "s" : ""} left`;
  }

  const hoursDifference = differenceInHours(targetDate, currentDate);
  if (hoursDifference > 0) {
    return `${hoursDifference} hour${hoursDifference > 1 ? "s" : ""} left`;
  }

  const minutesDifference = differenceInMinutes(targetDate, currentDate);
  if (minutesDifference > 0) {
    return `${minutesDifference} minute${minutesDifference > 1 ? "s" : ""} left`;
  }

  return "Market closed";
};

File: utils/calculateMarketClosing.ts [hunk 2]
```diff
@@ -1,38 +1,70 @@
...
   }
 
   return {
     newDate: targetDate,
     isClosed: isClosed,
-    daysLeft: daysLeft,
+    timeLeft: calculateTimeLeft(targetDate),
     formattedDate: formattedDate,
   };
 };
```
Comment (kacpergumieniuk) [inline utils/calculateMarketClosing.ts:41]: I think we should keep main return of the function at the very bottom of it, so for being consistent, we should put it below calculateTimeLeft func

File: utils/calculateMarketClosing.ts (hunk 1 above)
Comment (kacpergumieniuk) [reply utils/calculateMarketClosing.ts:70]: and what really that function does, it's giving us time left message, not just calculating it so we can think of changing the name to getTimeLeftMessage

File: utils/categorizeMarkets.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { calculateMarketClosing } from './calculateMarketClosing';
 export interface MarketWithDateType extends MarketType {
   newDate: Date;
   isClosed: boolean;
-  daysLeft: number;
+  timeLeft: string;
   formattedDate: string;
 }
```
Comment (kacpergumieniuk) [inline utils/categorizeMarkets.ts:9]: I think that we should change that prop from timeLeft to timeLeftMessage, as it's a message string, to keep the readability. Also, what you mentioned in your comment about keeping the daysLeft, i think we dont need it, but for future purposes of filtering markets by time, we should return some number representation of timeLeft.  Maybe we should just aslo return const diff = lifetime - blockHeight (called in some way of your choice, maybe timeDiff: number), to compare which market is older in the future?

**Review by kacpergumieniuk — CHANGES_REQUESTED** [review]

File: utils/calculateMarketClosing.ts (hunk 1 above)
Comment (salalo) [reply utils/calculateMarketClosing.ts:70]: Yeah, you're right, I wanted to omit difference calculations when it's not necessary, thus the nesting, but returning early works as well and it's more readable, good one. 

Also agree with changing the function name.

**Comment by salalo** [comment]
@kacpergumieniuk Btw, if you could add some formatting/linting config files it would be great, so it aligns all around the repo.

**Comment by kacpergumieniuk** [comment]
> @kacpergumieniuk Btw, if you could add some formatting/linting config files it would be great, so it aligns all around the repo.

yeah gonna do that soon

**Review by kacpergumieniuk — APPROVED** [review]

**Pull request merged**